### PR TITLE
toolchain: make toochain include path available to zephyr

### PR DESCRIPTION
**Pull request by nashif**
The include path is not available to Zephyr by default, this is contrary
to any other toolchain we use. For example unistd.h and friends are
available to Zephyr when building using xtools or embedded arm
toolchain, but not with the Zephyr SDK. This has been forcing us to
carry our own headers in the minimal c library.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

### CODE REVIEW

**Review by carlescufi — APPROVED** [review]
Looks definitely like an oversight when we were overwriting `TOOLCHAIN_INCLUDES`

**Review by ioannisg — APPROVED** [review]

**Comment by nashif** [comment]
That one failure we get is exactly the problem we have where things work with one toolchain and not the other because of the header confusion.

**Comment by galak** [comment]
> That one failure we get is exactly the problem we have where things work with one toolchain and not the other because of the header confusion.

So I'm not sure this makes sense.  We should be using the headers associated with the libc we are using.  So if we are building with newlib we'd use those, if we are building with minimal libc we should grab those headers.  It seems with this change we are mixing the two.

**Review by galak — CHANGES_REQUESTED** [review]
This change doesn't seem right.  We are mixing headers from two different libc which seems like just subject to problems.

**Comment by nashif** [comment]
> So I'm not sure this makes sense. We should be using the headers associated with the libc we are using.

We should be using the headers that come with the toolchain. Now the implementation of the libc (minimal lib or anything else) whether it is the one in the toolchain or a local one is a different story. This is how toolchains (other than the one we have in the zephyr sdk) are configured.


> This change doesn't seem right. We are mixing headers from two different libc which seems like just subject to problems.

Most of the headers we have as part minimal libc should be removed, we have been adding empty files and duplication of headers due to this bug.

**Comment by nashif** [comment]
btw, try building samples/subsys/ipc/openamp with gccarmemb, it will fail because the arm toolchain even with minimal libc enabled does provide the headers of the libc built with the toolchain, i.e. newlib.

This all sounds and looks wrong, but we have been adding ad-hoc headers to support features, for example posix APIs (in include/posix), empty headers such as lib/libc/minimal/include/sys/cdefs.h and incomplete headers in other places.

**Review by tyrantlizard77 — APPROVED** [review]

**Comment by galak** [comment]
> btw, try building samples/subsys/ipc/openamp with gccarmemb, it will fail because the arm toolchain even with minimal libc enabled does provide the headers of the libc built with the toolchain, i.e. newlib.
> 
> This all sounds and looks wrong, but we have been adding ad-hoc headers to support features, for example posix APIs (in include/posix), empty headers such as lib/libc/minimal/include/sys/cdefs.h and incomplete headers in other places.

Agree, this seems really wrong.  What happens if you configure gcc w/o newlib?  Do you still get a set of C headers?

**Comment by nashif** [comment]
> Agree, this seems really wrong. What happens if you configure gcc w/o newlib? Do you still get a set of C headers?

There are a few headers that come from gcc, without newlib being built, you will have among others:

```
float.h
gcov.h
iso646.h
stdalign.h
stdarg.h
stdatomic.h
stdbool.h
stddef.h
stdfix.h
stdint-gcc.h
stdint.h
stdnoreturn.h
tgmath.h
unwind.h
varargs.h
```

with newlib, you will get all the newlib headers obviously.

File: arch/posix/include/posix_cheats.h [hunk 1]
```diff
@@ -25,7 +25,6 @@
 
 #ifdef CONFIG_PTHREAD_IPC
 
-#define timespec zap_timespec
```
Comment (aescolar) [inline arch/posix/include/posix_cheats.h]: Why did you chose to remove this one? (it should work both with and without it)

**Review by aescolar — COMMENTED** [review]
Nothing against from my side.

**Comment by nashif** [comment]
will be resolved with sdk-ng

**Pull request closed without merging**